Feature/project independent work package create

ulferts opened the pull request:
Adds the project independent work package creation resource.

It adapts the schema, form and collection to reflect the added endpoint.

Also adds an available_projects endpoint to query for projects in which work packages can be created.
- [x] New endpoint and it's auxillary endpoint is documented in the api v3 documentation. Created by https://github.com/opf/openproject/pull/4267

https://community.openproject.com/work_packages/22847/activity

ulferts committed:
extract contract related methods of service into concern

ulferts committed:
remove offending is_*?

ulferts committed:
add scope to Type to query for those enabled in a project

ulferts committed:
improve CreateWorkPackageService to conform with new interface

ulferts committed:
Remove project#add_work_package

ulferts committed:
add a project independent endpoint for work package creation

ulferts committed:
move CreateFormResource to CreateProjectFormResource

ulferts committed:
explicitly require to avoid "still in tree" error

ulferts committed:
Handle empty create requests

ulferts committed:
add a project independent endpoint to create work packages

Includes the auxillary end points:
* form
* schema
* available_projects

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://ulferts-feature-project-independent-work-package-create.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://ulferts-feature-project-independent-work-package-create.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
Remove private annotation for descendents of FormRepresenter

oliverguenther committed:
Set project and default type for project-scoped form

This was the default for new forms before and no type was set (e.g., creating a child)
and should not be broken.
